Contents/Code/plex_api_helper.py

```diff
@@ -87,6 +87,30 @@ def setup_plexapi():
     return plex
 
 
+def is_field_locked(item, field_name):
+    # type: (any, str) -> bool
+    """
+    Check if the specified field is locked -- i.e. if the user has manually set a value for it and does not want
+    it to be updated by an automated process.
+
+    Parameters
+    ----------
+    item : any
+        The Plex item to check.
+    field_name : str
+        The name of the field to check.
+
+    Returns
+    -------
+    bool
+        True if the field is locked, False otherwise.
+    """
+    for field in item.fields:
+        if field.name == field_name:
+            return field.locked
+    return False
```

Comment:
```suggestion
    for field in item.fields:
        if field.name.lower() == field_name.lower():
            return field.locked
    Log.Exception('Plex {} {} does not contain the field "{}"'.format(item.type, item.ratingKey, raise AttributeError))
    return True  # we should return True if the field was not found
```

If the field is missing from the item, we shouldn't perform the modification in my opinion.

Also, if we convert to lowercase it will be a little more forgiving.

For the logging, not sure if this syntax is 100% correct but I think this may be better than just returning False. I don't think we should ever see this error in reality unless Plex changes field names. If we can't log and exception like this, then using "Error/Critical" is okay too I think.

Reply:
`field not in item.fields` does not mean the field is missing, it means the `<field` tag is missing. I know, it's confusing. Basically, Plex often doesn't store the `<field` tag if it's not locked. So you'll have a `<Video thumb="...">` that doesn't contain any `<Field name="thumb">`.

In other words:
- if the field is locked, then there is a `<Field name="..." locked="1" />`
- if the field is not locked, then there *may* be a `<Field name="..." locked="0" />`

Reply:
Okay that makes sense.